plugins/statscollector/plugin_impl_statscollector.go

```diff
@@ -126,45 +128,49 @@ func (p *Plugin) Init() error {
 	return nil
 }
 
-// AfterInit subscribes for monitoring of changes in ContainerIndex
-func (p *Plugin) AfterInit() error {
-	// watch containerIDX and remove gauges of pods that have been deleted
-	return p.Contiv.GetContainerIndex().Watch(p.PluginName, func(event containeridx.ChangeEvent) {
-		p.Lock()
-		defer p.Unlock()
-
-		var err error
-		if !event.Del {
-			return
-		}
+// processPodEvent processes pod delete events; it removes the deleted pod's
+// stats entry from the stats set reported to Prometheus.
+func (p *Plugin) processPodEvent(event containeridx.ChangeEvent) {
+	p.Lock()
+	defer p.Unlock()
 
-		nsmap, exists := p.podIfs[event.Value.PodNamespace]
-		if !exists {
-			return
-		}
-		ifs, exists := nsmap[event.Value.PodName]
-		if !exists {
-			return
-		}
-		for _, key := range ifs {
-			entry, exists := p.ifStats[key]
-			if exists && entry != nil {
-				// remove gauge with corresponding labels from each vector
-				for _, vec := range p.gaugeVecs {
-					vec.Delete(prometheus.Labels{
-						podNameLabel:       entry.podName,
-						podNamespaceLabel:  entry.podNamespace,
-						interfaceNameLabel: entry.data.Name,
-					})
-					if err != nil {
-						p.Log.Error(err)
-					}
+	var err error
+	if !event.Del {
+		return
+	}
+	nsmap, exists := p.podIfs[event.Value.PodName]
```

Comment:
first index should be namespace see https://github.com/contiv/vpp/blob/acc1665ab8af2c4e5d7a593f905b1ae6bee0052f/plugins/statscollector/plugin_impl_statscollector.go#L49

and the function where the values are actualy inserted 
https://github.com/contiv/vpp/blob/acc1665ab8af2c4e5d7a593f905b1ae6bee0052f/plugins/statscollector/plugin_impl_statscollector.go#L249-L255